Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h

```diff
@@ -0,0 +1,817 @@
+    CreateGroupingWorkspace groupAlg;
+    groupAlg.initialize();
+    groupAlg.setPropertyValue("InputWorkspace", inputWS);
+    groupAlg.setPropertyValue("GroupDetectorsBy", "All");
+    groupAlg.setPropertyValue("OutputWorkspace", groupWS);
+    groupAlg.execute();
+  }
+
+  void rebin(std::string params, bool preserveEvents = true) {
+    Rebin rebin;
+    rebin.initialize();
+    rebin.setPropertyValue("InputWorkspace", inputWS);
+    rebin.setPropertyValue("OutputWorkspace", inputWS);
+    rebin.setPropertyValue("Params", params);
+    rebin.setProperty("PreserveEvents", preserveEvents);
+    rebin.execute();
+    rebin.isExecuted();
+  }
+
+  void resamplex(int numHistoBins, bool preserveEvents = true) {
+    ResampleX resamplexAlg;
+    resamplexAlg.initialize();
+    resamplexAlg.setPropertyValue("InputWorkspace", inputWS);
+    resamplexAlg.setPropertyValue("OutputWorkspace", inputWS);
+    resamplexAlg.setProperty("NumberBins", numHistoBins);
+    resamplexAlg.setProperty("PreserveEvents", preserveEvents);
+    resamplexAlg.execute();
+  }
+
+  std::string createArgForNumberHistograms(double val, MatrixWorkspace_sptr ws,
+                                           std::string delimiter = ",") {
+    std::vector<std::string> vec;
+    for (size_t i = 0; i < ws->getNumberHistograms(); i++)
+      vec.push_back(boost::lexical_cast<std::string>(val));
+    std::string joined = boost::algorithm::join(vec, delimiter + " ");
+    return joined;
+  }
+
+  ITableWorkspace_sptr createMaskBinTable() {
+    maskBinTableWS = WorkspaceFactory::Instance().createTable();
+    maskBinTableWS->addColumn("str", "SpectraList");
+    maskBinTableWS->addColumn("double", "XMin");
+    maskBinTableWS->addColumn("double", "XMax");
+    TableRow row1 = maskBinTableWS->appendRow();
+    row1 << "" << 0.0 << 2000.0;
+    TableRow row2 = maskBinTableWS->appendRow();
+    row2 << "" << 10000.0 << xmax + 1000.0;
+    return maskBinTableWS;
+  }
+
+  void addPulseTimesForLogs() {
+    AddTimeSeriesLog logAlg;
+    std::string time, minute;
+    std::string prefix = "2010-01-01T00:";
+    for (int i = 0; i < 60; i++) {
+      minute =
+          std::string(2 - std::to_string(i).length(), '0') + std::to_string(i);
+      time = prefix + minute + "00";
+      logAlg.initialize();
+      logAlg.setPropertyValue("Workspace", inputWS);
+      logAlg.setPropertyValue("Name", "proton_charge");
+      logAlg.setPropertyValue("Time", time);
+      logAlg.setPropertyValue("Value", "100");
+    }
+    logAlg.execute();
+  }
+
+  void addFrequencyForLogs() {
+    AddSampleLog freqAlg;
+    freqAlg.initialize();
+    freqAlg.setPropertyValue("LogName", "Frequency");
+    freqAlg.setPropertyValue("LogText", "200.0");
+    freqAlg.setPropertyValue("LogUnit", "Hz");
+    freqAlg.setPropertyValue("LogType", "Number Series");
+    freqAlg.setPropertyValue("NumberType", "Double");
+    freqAlg.setPropertyValue("Workspace", inputWS);
+    freqAlg.execute();
+  }
+
+private:
```

Comment:
coding convention says member variables should start with `m_` to make them easier to spot. Also, initialized values should be defined as `int m_numBanks{1};` as done in newer C++.